driver/razermouse_driver.c

```diff
@@ -4666,6 +4686,81 @@ static void razer_mouse_init(struct razer_mouse_device *dev, struct usb_interfac
     dev->tilt_repeat = 33;
 }
 
+static int razer_battery_get_property(struct power_supply *ps,
+                                      enum power_supply_property property, union power_supply_propval *val)
+{
+    struct razer_mouse_device *dev = power_supply_get_drvdata(ps);
+
+    int ret = 0;
+    switch (property) {
+    case POWER_SUPPLY_PROP_PRESENT:
+        val->intval = 1;
+        break;
+    case POWER_SUPPLY_PROP_SCOPE:
+        val->intval = POWER_SUPPLY_SCOPE_DEVICE;
+        break;
+    case POWER_SUPPLY_PROP_CAPACITY:
+        val->intval = (razer_read_charge_level(dev) * 100) / 255;
```

Comment:
should be possible to remove the brackets, so use `razer_read_charge_level(dev) * 100 / 255` I think? You can also add a comment that it converts 0-255 level to 0-100 level.

Reply:
the unnecessary brackets is just for readability but i added the comment